### PR TITLE
Bugfix/get decimals from origin

### PR DESCRIPTION
**Pull request by coolwather**

### CODE REVIEW

File: src/components/transactions/modals/ClaimWRBTCModal.vue [hunk 1]
```diff
@@ -98,7 +98,7 @@
 <script>
 import { store } from '@/store'
 import { CLAIM_TYPES } from '@/constants/claimTypes'
-import { findNetworkByChainId } from '@/constants/networks'
+import { findNetworkByChainId, findNetworkById } from '@/constants/networks'
```
Comment (aguschirico) [inline src/components/transactions/modals/ClaimWRBTCModal.vue:101]: what's the difference between these 2 functions?

File: src/components/transactions/modals/ClaimWRBTCModal.vue [hunk 2]
```diff
@@ -189,19 +189,25 @@ export default {
           break
         }
         case this.claimTypes.CONVERT_TO_RBTC: {
-          this.processing = true
+          this.processing = true;
+          
+          const wBtcOriginDecimals =
```
Comment (aguschirico) [inline src/components/transactions/modals/ClaimWRBTCModal.vue:194]: did you test this with binance btc?

File: src/constants/networks.js [hunk 3]
```diff
@@ -213,6 +213,13 @@ export function findNetworkByChainId(chainId, crossToNetworkId) {
   )
 }
 
+export function findNetworkById(id) {
```
Comment (aguschirico) [inline src/constants/networks.js:216]: what's the difference between this function and the existing one? I think the naming is the only difference, functionally?

**Review by aguschirico — CHANGES_REQUESTED** [review]

File: src/components/transactions/modals/ClaimWRBTCModal.vue (hunk 1 above)
Comment (coolwather) [reply src/components/transactions/modals/ClaimWRBTCModal.vue]: Removed

File: src/constants/networks.js (hunk 3 above)
Comment (coolwather) [reply src/constants/networks.js]: Removed

File: src/components/transactions/modals/ClaimWRBTCModal.vue [hunk 4]
```diff
@@ -189,19 +189,25 @@ export default {
           break
         }
         case this.claimTypes.CONVERT_TO_RBTC: {
-          this.processing = true
+          this.processing = true;
+          
+          const wBtcOriginDecimals = 
+            findNetworkByChainId(this.transaction.networkId, this.transaction.destinationChainId).tokens.find(x => x.token === 'WBTC')?.decimals || 8;
+          
+          const weiDecimals = 18 - wBtcOriginDecimals;
```
Comment (pmprete) [inline src/components/transactions/modals/ClaimWRBTCModal.vue:197]: use destination chain token decimals

File: src/components/transactions/modals/ClaimWRBTCModal.vue [hunk 5]
```diff
@@ -189,19 +189,25 @@ export default {
           break
         }
         case this.claimTypes.CONVERT_TO_RBTC: {
-          this.processing = true
+          this.processing = true;
+          
+          const wBtcOriginDecimals = 
+            findNetworkByChainId(this.transaction.networkId, this.transaction.destinationChainId).tokens.find(x => x.token === 'WBTC')?.decimals || 8;
```
Comment (pmprete) [inline src/components/transactions/modals/ClaimWRBTCModal.vue:195]: remove || 8 instead throw an error if you could not find it

Comment (aguschirico) [inline src/components/transactions/modals/ClaimWRBTCModal.vue:195]: Just a suggestion here: I'd create a function called fromNetwork (see toNetwork function as an example)

**Review by aguschirico — APPROVED** [review]

**Pull request merged**